cavalier_contours/src/core/math/circle_circle_intersect.rs

```diff
@@ -25,7 +25,29 @@ where
     Overlapping,
 }
 
-/// Finds the intersects between two circles.
+/// Finds the intersects between two circles defined by the radius and center.
+///
+/// This function returns the geometric solution(s) for the intersection of two circles.
+/// The result will hold `NoIntersect`, if the circles are too far apart, `Overlapping`
+/// if the circles are similar in radii and center. In the other cases, the result will
+/// hold either a `TangentIntersect` with a single intersection point or `TwoIntersects`
+/// with two intersection points.
+///
+/// # Examples
+///
+/// ```
+/// # use cavalier_contours::core::traits::*;
+/// # use cavalier_contours::core::math::*;
+/// # use cavalier_contours::core::math::CircleCircleIntr::TangentIntersect;
+/// // Two tangent-intersecting circles of radius `1.0` with euclidean distance of `2.0`
+/// let r1 = 1.0;
+/// let c1 = Vector2::new(0.0, 0.0);
+/// let r2 = 1.0;
+/// let c2 = Vector2::new(0.0, 2.0);
+/// if let CircleCircleIntr::TangentIntersect{point:p} = circle_circle_intr(r1, c1, r2, c2) {
+///    assert_eq!(p, Vector2::new(0.0, 1.0));
+/// }
```

Comment:
This is really clean to read but will not execute any asserts if the intersect result is not the expected case of `CircleCircleIntr::TangentIntersect`, which may be a bit confusing to the reader. I'm not sure which is cleaner for the example but two ideas come to mind:
Add an else body to the if:
```rust
if let CircleCircleIntr::TangentIntersect{point:p} = circle_circle_intr(r1, c1, r2, c2) {
    assert_eq!(p, Vector2::new(0.0, 1.0));
} else {
    unreachable!("expected tangent intersect");
}
```
Or use the `matches!` macro like the following:
```rust
let intr_result = circle_circle_intr(r1, c1, r2, c2);
assert!(matches!(intr_result, CircleCircleIntr::TangentIntersect{ point } if point == Vector2::new(0.0, 1.0));
```

Maybe there's a better third option...

Reply:
I, personally, find the former choice easier to read due to the fact that the `if let` clearly shows the matching arm. Also the nested functions are always harder to read. And as lines do not cost a dime, I always go with the solution that is easier to digest.

Reply:
I will embed your suggestions this week.